src/task/syscall.rs

```diff
@@ -32,21 +37,122 @@ const MSR_STAR: u32 = 0xc000_0081;
 const MSR_LSTAR: u32 = 0xc000_0082;
 const MSR_SFMASK: u32 = 0xc000_0084;
 
+#[repr(u32)]
+#[derive(Debug)]
+pub enum SystemCalls {
+    Exit = 0,
+    Log = 1,
+    Sleep = 2,
+}
+
+fn map_to_syscall(index: u32) -> Option<SystemCalls> {
+    match index {
+        1 => Some(SystemCalls::Exit),
+        2 => Some(SystemCalls::Log),
+        3 => Some(SystemCalls::Sleep),
+        _ => None,
+    }
+}
```

Comment:
Can we do this via `TryFrom`? i.e:

```rust
impl TryFrom<u32> for SystemCalls {
    type Error = SvsmError;
    fn try_from(index: u32) -> Result<SystemCalls, Self::Error> {
        ...
    }
}
```

Reply:
Yes, I changed this and added the new error InvalidSystemCall